xnmt/translator.py

```diff
@@ -131,7 +132,9 @@ def generate(self, src, idx, src_mask=None, forced_trg_ids=None):
       # In case of reporting
       if self.report_path is not None:
         src_words = [self.reporting_src_vocab[w] for w in sents]
-        trg_words = [self.trg_vocab[w] for w in output_actions[1:]]
+        # next line seemed to truncate first trg elt in the seq (in forced decoding mode at least)
+        # `output_actions[1:]` replaced by `output_actions[0:]`
+        trg_words = [self.trg_vocab[w] for w in output_actions[0:]]
```

Comment:
This should not be `[0:]`, you can just remove this.

Reply:
Thank you very much for taking a look. But if I put back `output_actions[1:]`, then the target is truncated from its first unit, and the attention matrix's shape doesn't match (len(src), len(trg)) anymore. Is that what you meant?

Reply:
No, I meant remove `[0:]`, leaving only `output_actions`. `[0:]` is a noop and just makes the code more complicated.

Reply:
Oh sorry, yes. So silly of me. Done.